src/main/java/graphql/nadel/ServiceExecutionParameters.java

```diff
@@ -2,40 +2,98 @@
 
 import graphql.PublicApi;
 import graphql.language.Document;
+import graphql.language.FragmentDefinition;
+import graphql.language.OperationDefinition;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static graphql.Assert.assertNotNull;
 
 
 @PublicApi
 public class ServiceExecutionParameters {
 
     private final Document query;
+    private final Object context;
+    private final Map<String, Object> variables;
+    private final Map<String, FragmentDefinition> fragments;
+    private final OperationDefinition operationDefinition;
 
-    private ServiceExecutionParameters(Document query) {
-        this.query = query;
+    private ServiceExecutionParameters(Document query, Object context, Map<String, Object> variables, Map<String, FragmentDefinition> fragments, OperationDefinition operationDefinition) {
+        this.query = assertNotNull(query);
+        this.variables = assertNotNull(variables);
+        this.fragments = assertNotNull(fragments);
+        this.operationDefinition = assertNotNull(operationDefinition);
+        this.context = context;
     }
 
     public Document getQuery() {
         return query;
     }
 
-    public static Builder newDelegatedExecutionParameters() {
+    public Object getContext() {
+        return context;
+    }
+
+    public Map<String, Object> getVariables() {
+        return variables;
+    }
+
+    public Map<String, FragmentDefinition> getFragments() {
+        return fragments;
+    }
+
+    public OperationDefinition getOperationDefinition() {
+        return operationDefinition;
+    }
+
+    public OperationDefinition.Operation getOperation() {
```

Comment:
Do we need this ?  Seems to me that we can remove this if there is no special logic to get operation.

Reply:
Right now - no - but in the future a service call might want to know if its a Mutation / Query